sklearn/linear_model/_coordinate_descent.py

```diff
@@ -140,43 +143,36 @@ def _alpha_grid(
             "your estimator with the appropriate `alphas=` "
             "argument."
         )
-    n_samples = len(y)
-
-    sparse_center = False
-    if Xy is None:
-        X_sparse = sparse.issparse(X)
-        sparse_center = X_sparse and fit_intercept
-        X = check_array(
-            X, accept_sparse="csc", copy=(copy_X and fit_intercept and not X_sparse)
+    if Xy is not None:
+        Xyw = Xy
+    else:
+        X, y, X_offset, _, _ = _preprocess_data(
+            X,
+            y,
+            fit_intercept,
```

Comment:
```suggestion
            fit_intercept=fit_intercept,
```

Reply:
Yes whoops, I think we did it together but then I did not re-add it into the commit